sdk/eventhub/Azure.Messaging.EventHubs/tests/Processor/EventProcessorTests.cs

```diff
@@ -143,68 +110,130 @@ public void ConstructorClonesOptions()
+        ///    Verifies functionality of the <see cref="EventProcessor" />
         ///    constructor.
         /// </summary>
         ///
         [Test]
         public void ConstructorCreatesTheIdentifier()
         {
-            EventHubClient mockClient = Mock.Of<EventHubClient>();
-            Func<PartitionContext, PartitionProcessor> mockFactory = Mock.Of<Func<PartitionContext, PartitionProcessor>>();
-            PartitionManager mockPartitionManager = Mock.Of<PartitionManager>();
-
-            var eventProcessor = new EventProcessor<PartitionProcessor>(EventHubConsumer.DefaultConsumerGroupName, mockClient, mockFactory, mockPartitionManager);
+            var eventProcessor = new EventProcessor("consumerGroup", Mock.Of<EventHubClient>(), Mock.Of<PartitionManager>());
 
             Assert.That(eventProcessor.Identifier, Is.Not.Null);
             Assert.That(eventProcessor.Identifier, Is.Not.Empty);
         }
 
+        /// <summary>
+        ///    Verifies functionality of the <see cref="EventProcessor.StartAsync" />
+        ///    method.
+        /// </summary>
+        ///
+        [Test]
+        public void StartAsyncValidatesProcessEventsAsync()
+        {
+            EventProcessor processor = new EventProcessor("consumerGroup", Mock.Of<EventHubClient>(), Mock.Of<PartitionManager>());
+            processor.ProcessExceptionAsync = (context, exception) => Task.CompletedTask;
+
+            Assert.That(async () => await processor.StartAsync(), Throws.InstanceOf<InvalidOperationException>().And.Message.Contains(nameof(EventProcessor.ProcessEventsAsync)));
+        }
+
+        /// <summary>
+        ///    Verifies functionality of the <see cref="EventProcessor.StartAsync" />
+        ///    method.
+        /// </summary>
+        ///
+        [Test]
+        public void StartAsyncValidatesProcessExceptionAsync()
+        {
+            EventProcessor processor = new EventProcessor("consumerGroup", Mock.Of<EventHubClient>(), Mock.Of<PartitionManager>());
+            processor.ProcessEventsAsync = (context, events) => Task.CompletedTask;
+
+            Assert.That(async () => await processor.StartAsync(), Throws.InstanceOf<InvalidOperationException>().And.Message.Contains(nameof(EventProcessor.ProcessExceptionAsync)));
+        }
+
+        /// <summary>
+        ///    Verifies functionality of the <see cref="EventProcessor.StartAsync" />
+        ///    method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task StartAsyncStartsTheEventProcessorWhenProcessingHandlerPropertiesAreSet()
+        {
+            EventProcessor processor = new EventProcessor("consumerGroup", Mock.Of<EventHubClient>(), Mock.Of<PartitionManager>());
+
+            processor.ProcessEventsAsync = (context, events) => Task.CompletedTask;
+            processor.ProcessExceptionAsync = (context, exception) => Task.CompletedTask;
+
+            Assert.That(async () => await processor.StartAsync(), Throws.Nothing);
```

Comment:
Does `StartProcessingEvents` still need to be async?

Reply:
We are still awaiting the `RunningTaskSemaphore` in `StartAsync`. We use it to avoid race conditions with `StopAsync` and other calls to itself.